### PR TITLE
net: Move SocketSendData lock annotation to header

### PR DESCRIPTION
**Pull request by maflcko**
Lock annotations must be in the header, otherwise the will have limited or no effect

### CODE REVIEW

**Comment by ajtowns** [comment]
Concept ACK.

If you're moving things around in net.h to add lock annotations, might be worth moving `NetEventsInterface` to after the `CNode` definition, and making it `virtual bool SendMessages(CNode* pnode) EXCLUSIVE_LOCKS_REQUIRED(pnode->cs_sendProcessing) = 0;`.

Moving `IsPeerAddrLocalGood` and `AdvertiseLocal` later as well would allow removing the `class CNode;` forward declaration entirely.

It would be nice if there were some easy way to review pointer-becomes-reference changes.

**Comment by MarcoFalke** [comment]
> It would be nice if there were some easy way to review pointer-becomes-reference changes.

Apart from `--word-diff-regex=.`?

> SendMessages ...

Thanks, done

File: src/net.h [hunk 1]
```diff
@@ -180,1041 +180,1040 @@ enum class ConnectionType {
...
+     * A vector of -bind=<address>:<port>=onion arguments each of which is
+     * an address and port that are designated for incoming Tor connections.
+     */
+    std::vector<CService> m_onion_binds;
 
-    /** Whether this peer is an inbound onion, e.g. connected via our Tor onion service. */
-    bool IsInboundOnion() const { return m_inbound_onion; }
+    friend struct CConnmanTest;
+    friend struct ConnmanTestMsg;
 };
```
Comment (jnewbery) [inline src/net.h:1216]: If you touch this again, add a blank line after the class declaration.

**Review by jnewbery — COMMENTED** [review]
ACK fad2e1f267d60afe9799e431233f54f02d14e8e0

Nice tidy-up.

**Review by theStack — APPROVED** [review]
ACK fad2e1f267d60afe9799e431233f54f02d14e8e0
With the suggestions in the commit messages on how to best show the diffs, the review was quite straight-forward and painless. 👌

**Comment by ajtowns** [comment]
ACK fad2e1f267d60afe9799e431233f54f02d14e8e0

**Comment by MarcoFalke** [comment]
Rebased, should be trivial to re-ACK with git range-diff or from scratch

File: src/net.h (hunk 1 above)
Comment (MarcoFalke) [reply src/net.h:1216]: thanks, done

**Comment by jnewbery** [comment]
utACK fa210689e2

**Pull request merged**